src/lib/util/cdrom.cpp

```diff
@@ -1493,17 +1495,17 @@ uint64_t cdrom_file::get_file_size(std::string_view filename)
 int cdrom_file::tokenize( const char *linebuffer, int i, int linebuffersize, char *token, int tokensize )
 {
 	int j = 0;
-	int singlequote = 0;
-	int doublequote = 0;
+	bool singlequote = false;
+	bool doublequote = false;
 
 	while ((i < linebuffersize) && isspace((uint8_t)linebuffer[i]))
 	{
 		i++;
 	}
 
-	while ((i < linebuffersize) && (j < tokensize))
+	while ((i < linebuffersize) && (j < tokensize) && (linebuffer[i] != '\0'))
```

Comment:
How does `linebuffer[i] != '\0'` happen?  Is `linebuffersize` the size of the buffer rather than the size of its content?  The code seemed to be treating it as the size of the content.

Reply:
`linebuffersize` is just `sizeof(linebuffer)` so it's the entire buffer size rather than its contents. So without the null byte check, the entire buffer is processed every time instead of just ending when reasonable.

https://github.com/mamedev/mame/blob/1565fb9ba6e2543558553131dc2e31dedfc41213/src/lib/util/cdrom.cpp#L1411-L1422

Reply:
Eugh, if linebuffer is a pointer, bad things will happen.  Will it work if you change it to `tokenize( linebuffer, i, std::size(linebuffer), token, std::size(token) );` so it will give an error if `linebuffer` and `token` aren’t array-like?

Reply:
Yeah, it works from some quick testing.